### ISSUE TITLE
labels for each object prior

### ISSUE THREAD

**Issue by SlowMonk** [tensorflow/tensorflow#42716]
I'm building ssd(object detection) with pytorch-> tensorflow<br/>

let say I have 

    object_for_each_prior
    <tf.Tensor: shape=(8,), dtype=int64, numpy=array([2, 2, 0, 1, 0, 0, 1, 1])>
    
    labels
    <tf.Tensor: shape=(3,), dtype=int32, numpy=array([ 20,  10, 5], dtype=int32)>

and want to represent the 

    label_for_each_prior = labels[object_for_each_prior]  # (8732)

expected output should be 

    tensor([ 5,  5,  0, 10,  0,  0, 10, 10], device='cuda:0')

**Comment by Saduf2019** [comment]
@SlowMonk 
Please fill in [issue template](https://github.com/tensorflow/tensorflow/issues/new/choose) for us to analyse the issue.
